presto-client/src/utils.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Parses a JSON including bigger numbers into BigInts
+ * This function checks if JSON.parse reviver callback has a context parameter
+ * and falls back onto the default parsing if not.
+ * See also:
+ * - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#browser_compatibility
+ * - https://github.com/tc39/proposal-json-parse-with-source
+ * @param _ Key
+ * @param value Parsed value
+ * @param context Context with source text
+ * @returns Parsed object with BigInts where required
+ */
+export function parseWithBigInts(_: string, value: unknown, context: { source: string }) {
+  if (!context) return value // Context is not available, fallback to default parse
+  const { source } = context
+  if (!source) return value // Source is not available, fallback to default parse
```

Comment:
This can be simplified as:
```suggestion
  if (!context?.source) return value
```